### ISSUE TITLE
Agent Start Fails with `ErrorQUICClientInternal: Failed to send data on the QUICSocket`

### ISSUE THREAD

**Issue by CryptoTotalWar** [MatrixAI/Polykey-CLI#219]
### Describe the bug

When starting an agent it fully starts, it makes a few connections but when it hits the ipv6 connections it fails with the following error. 

We got this error after adding a console.e log in the function ` handleEventQUICConnectionSend = async (evt) => ` inside QUICCLIENT.js 

```
Error: send EHOSTUNREACH 2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314
    at doSend (node:dgram:717:16)
    at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
    at afterDns (node:dgram:663:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -65,
  code: 'EHOSTUNREACH',
  syscall: 'send',
  address: '2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649',
  port: 1314
}
[2600:1f16:1f71:7c00:4aa5:9c9d:57f5:a649:1314].QUICClient:ErrorQUICClientInternal: QUIC Client internal error - Failed to send data on the QUICSocket
Error: send EHOSTUNREACH 2406:da1c:67c:c000:3ed6:cc4b:b6e5:3745:1314
    at doSend (node:dgram:717:16)
    at defaultTriggerAsyncIdScope (node:internal/async_hooks:464:18)
    at afterDns (node:dgram:663:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -65,
  code: 'EHOSTUNREACH',
  syscall: 'send',
  address: '2406:da1c:67c:c000:3ed6:cc4b:b6e5:3745',
  port: 1314
}
[2406:da1c:67c:c000:3ed6:cc4b:b6e5:3745:1314].QUICClient:ErrorQUICClientInternal: QUIC Client internal error - Failed to send data on the QUICSocket
ErrorQUICClientInternal: Failed to send data on the QUICSocket
```

### To Reproduce
1. doing a PK agent start 
2. it fully starts 
3. it makes a few connections
4. it fails making an ipv6 connection 

### Expected behaviour
Should not fail. 

### Screenshots
[Failed_to_connect_to_ipv6.txt](https://github.com/user-attachments/files/16064718/Failed_to_connect_to_ipv6.txt)

### Platform (please complete the following information)
 - Device: MAC OS 
 - PK Version `["0.5.5","1.6.4","1","1"]`

### Additional context
Did some debugging with Brian. 

### Notify maintainers
@tegefaulkes

**Comment by tegefaulkes** [comment]
`EHOSTUNREACH` is an odd error we haven't seen before. It seems to only happen on @CryptoTotalWar 's machine. We already special case similar errors such as `EINVAL` or `ENETUNREACH` so we may just need to do the same here. I just need to understand why we're getting this error in this case.

**Comment by CryptoTotalWar** [comment]
In the meantime, Pablo will have to start Polykey this way because his ipv6 is not working `polykey agent start --agent-host 0.0.0.0`

**Comment by tegefaulkes** [comment]
`EHOSTUNREACH` indicates `No route to the host is available.` Usually you wouldn't get this if you can't find an address. AFAIK it requires *something* to respond with an ICMP packet indicating it. In anycase, it's a problem relating to the specific connection so I'm adding it to the list of errors we dispatch back to the connection.

This will prevent js-quic from crashing outright but keep in mind that the underlying problem still exists. It's pretty likely a problem with your network setup though since during testing you couldn't ping a common google IPv6 address.

**Comment by tegefaulkes** [comment]
I'm creating a fix in `js-quic`. This will not be resolved until the following conditions are met.

- [X] 1\. Fix in js-quic is applied and released
- [X] 2\. js-quic dependency in CLI is updated with the fix.
- [X] 3\. We test on @pablo.padillo machine with the same conditions. IPv6 connections still won't work but the CLI shouldn't crash anymore.

**Comment by CryptoTotalWar** [comment]
<img width="338" alt="image" src="https://github.com/user-attachments/assets/fec0f977-414d-4bb1-85d6-76a4ddd22e93">

FYI I got a `2 packages looking for funding` warning upon updating to the latest cli version

**Comment by CryptoTotalWar** [comment]
@tegefaulkes Polykey is not crashing for me anymore but I am also not testing from the same network environment as I was last time. 

I just ran `ipconfig` and can confirm that my current network env is configured to use both IPv4 and IPv6 addresses. 

I then ran `polykey agent status` which shows that my `clientHost` is using IPv4 while the `agentHost` is using IPv6

Are there any other tests you would like for me to run?
